bangc-ops/mlu_op.h

```diff
@@ -2657,6 +2657,105 @@ mluOpYoloBox(mluOpHandle_t handle,
+ * @param[in] num_points
+ * The number of voxels.
+ * @param[in] num_channels
+ * The number of channels for voxels features.
+ * @param[in] num_voxel_x
+ * The number of voxels for dimX.
+ * @param[in] num_voxel_y
+ * The number of voxels for dimY.
+ * @param[in] num_voxel_z
+ * The number of voxels for dimZ.
+ * @param[in] geom_xyz_desc
+ * The descriptor of the tensors. For detailed information, see ::mluOpTensorDescriptor_t.
+ * @param[in] geom_xyz
+ * Pointer to the MLU memory that stores the input tensor.The features shape (batch_size, num_points, 3).
+ * @param[in] input_features_desc
+ * The descriptor of the tensors. For detailed information, see ::mluOpTensorDescriptor_t.
+ * @param[in] input_features
+ * Pointer to the MLU memory that stores the input tensor.The features shape (batch_size, num_points, num_channels).
+ * @param[in] output_features_desc
+ * The descriptor of the tensors. For detailed information, see ::mluOpTensorDescriptor_t.
+ * @param[out] output_features
+ * Pointer to the MLU memory that stores the output tensor.The features shape (batch_size, num_voxel_y, num_voxel_x, num_channels).
+ * @param[in] pos_memo_desc
+ * The descriptor of the tensors. For detailed information, see ::mluOpTensorDescriptor_t.
+ * @param[out] pos_memo
+ * Pointer to the MLU memory that stores the output tensor.The features shape (batch_size, num_points, 3).
+ *
+ * @par Return
+ * - ::MLUOP_STATUS_SUCCESS, ::MLUOP_STATUS_BAD_PARAM, ::MLUOP_STATUS_NOT_SUPPORTED
+ *
+ * @par Data Type
+ * - Data types of input tensors and output tensor must be the same.
+ * - The supported data types of input and output tensors are as follows:
+ *
+ *   - geom_xyz tensor: int.
+ *   - input_features tensor: float.
+ *   - output_features tensor: float.
+ *   - pos_memo tensor: int.
+ *
+ * @par Data Layout
+ *  - The supported data layouts of \b geom_xyz, \b input_features, \b output_features and \b pos_memo are
+ *    as follows:
+ *
+ *   - Input tensor:
+ *     - geom_xyz tensor: \p MLUOP_LAYOUT_ARRAY.
+ *     - input_features tensor: \p MLUOP_LAYOUT_ARRAY.
+ *   - Output tensor:
+ *     - output_features tensor: \p MLUOP_LAYOUT_ARRAY.
+ *     - pos_memo tensor: \p MLUOP_LAYOUT_ARRAY.
+ *
+ *  @par Limitation
+ *  - The geom_xyz tensor, input_features tensor and pos_memo tensor must be 3-D.
+ *  - The output_features tensor must be 4-D.
+ *  - The shape of \b geom_xyz should be [batch_size, num_points, 3].
+ *  - The shape of \b input_features should be [batch_size, num_points, num_channels].
+ *  - The shape of \b output_features should be [batch_size, num_voxel_y, num_voxel_x, num_channels].
+ *  - the shape of \b pos_memo should be [batch_size, num_points, 3].
+ *  - The \b batch_size , \b num_points , \b num_channels , \b num_voxel_x , \b num_voxel_y and \b num_voxel_z should be larger than or equal to 1.
+ *  - The output \b pos_memo needs to set the initial value before calling the operator, and initialize it to a negative number, such as -1.
```

Comment:
You need to set the initial value for the output \b pos_memo before calling the operator, and initialize it to a negative number, such as -1.